### ISSUE TITLE
Deploy on maven central?

### ISSUE THREAD

**Issue by victornoel** [jidesoft/jide-oss#27]
Hi,

There seems to be only old versions available on maven central, is this intended?

See: http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.jidesoft%22%20AND%20a%3A%22jide-oss%22

**Comment by jidesoft** [comment]
Not intended but we have trouble uploading the releases. Tried a few times but couldn't do it. Seems to be something wrong with the gradle script. Not sure if there is a way to manually upload the files.

**Comment by akazancev** [comment]
This is kind of important issue. Do you have any ETA for this?
Thanks,
-Alex

**Comment by jidesoft** [comment]
We tried but couldn't get it working. So no ETA for now. Please import the jars manually.

**Comment by akazancev** [comment]
Thank you! 
Very sad :(( We cannot do manual import due to our company policies. Only open source projects published in maven central can be used.

**Comment by victornoel** [comment]
@jidesoft could you share with us what you tried to do and what failed? maybe someone can help :)

**Comment by lambdas** [comment]
Hi, guys! Any success with this? It's really inconvenient to be unable to grab the library from maven.

**Comment by swimmesberger** [comment]
Would appreciate a maven central deployment of the current version.

**Comment by bourgesl** [comment]
I fixed the maven build in PR:
https://github.com/jidesoft/jide-oss/pull/32

**Comment by illumi** [comment]
@jidesoft can you merge #32 and perform a deploy?

**Comment by jidesoft** [comment]
Just did

**Comment by illumi** [comment]
@jidesoft 3.7.x still not available on central. Any chance..?

**Comment by bourgesl** [comment]
As @jidesoft is not making any progress on publishing binary artifacts on maven central, I will release my own build (jar files) on github.
It is FOSS, do what you want with good source code...

**Comment by bourgesl** [comment]
Here is my unofficial build:
https://github.com/bourgesl/jide-oss/releases/tag/3.7.4-b20181214

**Comment by lambdas** [comment]
@bourgesl Maybe you can push your fork to maven and update it with every release.. That would be awesome.

**Comment by bourgesl** [comment]
Even if jide-oss is free code, I can not publish on maven central instead.
Sorry, I just published 1 binary release to let you enjoy it, I will not release jide-oss releases myself, it is their project & releaseq.

**Comment by jidesoft** [comment]
> Even if jide-oss is free code, I can not publish on maven central instead.
> Sorry, I just published 1 binary release to let you enjoy it, I will not release jide-oss releases myself, it is their project & releaseq.
Hi Bourgesl,

Do you think you can publish it on the maven repository if I made you an admin for this project?

-David

**Comment by illumi** [comment]
Hi David,

GitHub and maven central are two different systems. Making someone an admin on GitHub doesn't mean they can deploy in the name of jidesoft.

Are you saying Jidesoft cannot or has lost the ability or knowledge to deploy artifacts?

**Comment by jidesoft** [comment]
We couldn't get it working.

**Comment by bourgesl** [comment]
Sorry, I do not understand.

@jidesoft you are not able to publish on maven central (nexus account) ?
I do have an account (for the marlin renderer) but never pushed (properly) artifacts on nexus, lack of time.

If you want me to help you, I could but please contact me directly via twitter DM (see https://github.com/bourgesl)

**Comment by Joerg-Schoemer** [comment]
Hi all,

you use gradle to publish to a nexus on a maven project? This sounds strange to me!

Maven supplies a release-plugin to publish to maven repositories.
I've build a pull request to migrate completely to the maven structure. Now it should be easy to publish with maven. For the first time the maven project supports only java8. 

Kind regards
Jörg

**Comment by mokun** [comment]
Hi David, 

Could you configure this fine library as a maven artifact using [jitpack](https://jitpack.io/) in GitHub? 

That way, we can use the latest version with bugs fixed. 

Thanks!

**Comment by mokun** [comment]
Anyone is working on deploying the latest jide-oss to a repo ?

**Comment by akazancev** [comment]
@mokun I doubt:((
I think most people are working on getting rid of jide-oss dependency completely, especially due to Java 11 incompatibility of published versions.

**Comment by apjanke** [comment]
Hi folks,

I think I have Maven Central publishing working under the `net.janklab.opp` group. (I'm doing this as part of the [Janklab-OPP project](https://github.com/apjanke/janklab-opp), which aims to make Maven Central builds of libraries used by Matlab but which don't do their own Maven Central publishing.) I've published a SNAPSHOT of jide-oss 3.7.4. Anyone want to test it out? It'd be nice to have confirmation this is working for other users before I publish the non-snapshot release.

To pull it in, you need to enable the snapshot repo in your project's POM:

```xml
  <repositories>
    <repository>
      <id>oss.sonatype.org-snapshot</id>
      <url>http://oss.sonatype.org/content/repositories/snapshots</url>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
    </repository>
  </repositories>
```

Then reference the net.janklab.opp:jide-oss artifact:

```xml
  <dependencies>
    <dependency>
      <groupId>net.janklab.opp</groupId>
      <artifactId>jide-oss</artifactId>
      <version>3.7.4-SNAPSHOT</version>
    </dependency>
  </dependencies>
```

You can see the tweaks I made to the jide-oss pom.xml here:
* https://github.com/apjanke/janklab-opp/blob/bd7de70c39e3d980ea4c7a4c37eab039c796371d/jide-oss/jide-oss-3.7.4/project/jide-oss-3.7.4/pom.xml
* https://github.com/apjanke/janklab-opp/commit/2446b03760fd48353dcb1537fe66a04bb3f5e351
* https://github.com/apjanke/janklab-opp/commit/59341f9eea3472a26c373957471ae23fda5cf2a2
* https://github.com/apjanke/janklab-opp/commit/74acde8cff1cced135784bf239ebde7a6ffa8779
* https://github.com/apjanke/janklab-opp/commit/bd7de70c39e3d980ea4c7a4c37eab039c796371d

**Comment by mokun** [comment]
@apjanke , 

So, for 3.7.4, use `<version>3.7.4</version>` (without the word -SNAPSHOT), right ?

But for 3.7.6, use `<version>3.7.6-SNAPSHOT</version>` (with the word -SNAPSHOT), right ?

**Comment by mokun** [comment]
@apjanke ,

How come your 3.7.4 has a lot of files missing ? 

For instance, when I try to look for `LookAndFeelFactory.java` as in Eclipse IDE, it won't let me. 

It turns out that this java class and a lot of other class files are not in this `jide-oss-3.7.4.jar` jarfile that you distributed in maven.

https://github.com/jidesoft/jide-oss/blob/master/src/com/jidesoft/plaf/LookAndFeelFactory.java

When I use 3.6.18 and downloaded those files locally to my machines, note their sizes are big.
![image](https://user-images.githubusercontent.com/1168584/89742017-ffaa3000-da4a-11ea-82a7-b39bb1f22747.png)

When I switch to your distribution and use 3.7.4 and downloaded them locally, note their sizes are much smaller.

![image](https://user-images.githubusercontent.com/1168584/89742062-79421e00-da4b-11ea-8d10-77342473116f.png)

So, did jide stripped off those size intentionally in newer versions ?

**Comment by apjanke** [comment]
> So, did jide stripped off those size intentionally in newer versions ?

Well, uh-oh: I have no idea! I'll look in to this.

**Comment by tferr** [comment]
@apjanke, since thank you for working on this! I can confirm the jar contains no compiled classes, just resource files, such as icons and such. Did you manage to find out what the issue is?

**Comment by apjanke** [comment]
No, sorry! I've gotten really busy at the day job.

I'll try to get to this soon since I know people are waiting on it.

Cheers,
Andrew

**Comment by mokun** [comment]
@apjanke 

Well obviously it doesn't compile correctly as you know.

But since now you're the one who knows how to post it in maven repo, we're hoping you can figure it out.

Best regards !

**Comment by sfuerte** [comment]
as it's been dragging already for almost 3 years, is it possible to be released via Github packages at least? 

https://docs.github.com/en/free-pro-team@latest/packages/using-github-packages-with-your-projects-ecosystem/configuring-apache-maven-for-use-with-github-packages#installing-a-package

Thank you!

**Comment by mokun** [comment]
@sfuerte 

It'll be awesome if someone who are familiar with the process of releasing it as a github packages can spearhead this.

**Comment by sfuerte** [comment]
@mokun 
I believe the documentation is rather explanatory and clear. But if need any hand, sure thing, send me a PM.

On the another hand, it looks like there is a problem with 3.7.x release builds in general, logged it in a separate issue - https://github.com/jidesoft/jide-oss/issues/48

**Comment by jidesoft** [comment]
The compilation error is one of the reason we gave up deploying on maven.

**Comment by sfuerte** [comment]
> The compilation error is one of the reason we gave up deploying on maven. 

@jidesoft , does it mean that `jide-oss` is simply an abandoned project? 

Your Maven POM is a bit outdated and hence doesn't compile properly. And can be fixed quite easily. 

But the source of the problem isn't that (as Ant build produces errors as well) but rather project's dependency on the Windows LAF which is not included anymore in non-Windows JDK releases, see more at https://bugs.openjdk.java.net/browse/JDK-6461834

https://github.com/jidesoft/jide-oss/issues/48 is pretty much a blocker for Maven builds and releases.

**Comment by jidesoft** [comment]
@sfuerte Swing is dying so we are not very actively working on this project except for bug fixes.

**Comment by sfuerte** [comment]
@jidesoft , will #48 be considered as a bug that requires a fix?

**Comment by DevCharly** [comment]
I've built and published latest jide-oss version to **Maven Central**:
https://search.maven.org/artifact/com.formdev/jide-oss

~~~
<dependency>
  <groupId>com.formdev</groupId>
  <artifactId>jide-oss</artifactId>
  <version>3.7.11.1</version>
</dependency>
~~~

This was built on GitHub Actions using Gradle.
Project is here: https://github.com/JFormDesigner/jide-oss
